server/handlers_linkpreview.go

```diff
@@ -77,6 +102,44 @@ func handleLinkPreviewThumbnail(db *sql.DB, logger *zap.Logger) http.HandlerFunc
 	}
 }
 
+func handleLinkPreviewFavicon(db *sql.DB, logger *zap.Logger) http.HandlerFunc {
```

Comment:
Same as my comment above. We shouldn't copy & paste this much. The only difference with `handleLinkPreviewThumbnail` is that you need to call `getFaviconPayload` instead of `getThumbnailPayload`. You can create a common function that receives a function fulfilling the signature of both (which are identical).